avalon/tools/cbsceneinventory/app.py

```diff
@@ -224,6 +239,199 @@ def show_remove_warning_dialog(self, items):
         self.data_changed.emit()
 
 
+class SwitchAssetDialog(QtWidgets.QDialog):
+    """Widget to support asset switching"""
+
+    switched = QtCore.Signal()
+
+    def __init__(self, parent=None, items=None):
+        QtWidgets.QDialog.__init__(self, parent)
+
+        self._items = items
+
+        self._assets_box = None
+        self._subsets_box = None
+        self._representations_box = None
+
+        input_layout = QtWidgets.QHBoxLayout()
+        input_layout.setDirection(QtWidgets.QBoxLayout.RightToLeft)
+
+        accept_icon = qta.icon("fa.check-square", color="white")
+        accept_btn = QtWidgets.QPushButton()
+        accept_btn.setIcon(accept_icon)
+        accept_btn.setFixedWidth(24)
+        accept_btn.setFixedHeight(24)
+
+        input_layout.addWidget(accept_btn)
+
+        self._input_layout = input_layout
+        self._accept_btn = accept_btn
+
+        self.setLayout(input_layout)
+        self.setWindowTitle("Switch selected items ...")
+
+        self.connections()
+
+        self.refresh()
+
+    def connections(self):
+        self._accept_btn.clicked.connect(self._on_accept)
+
+    def refresh(self):
+
+        assets = self._get_assets()
+        self._assets_box = self._create_combo_box(assets)
+
+        subsets = self._get_subsets()
+        self._subsets_box = self._create_combo_box(subsets)
+
+        representations = self._get_representations()
+        self._representations_box = self._create_combo_box(representations)
+
+        self._input_layout.addWidget(self._representations_box)
+        self._input_layout.addWidget(self._subsets_box)
+        self._input_layout.addWidget(self._assets_box)
+
+    def _create_combo_box(self, items):
+        """Create a combobox with auto completion, first item will be "----"
+
+        Args:
+            items (iterable): list of document names
+
+        Returns:
+            QtWidgets.QComboBox
+
+        """
+
+        combobox = QtWidgets.QComboBox()
+        completer = QtWidgets.QCompleter(items)
+        combobox.setCompleter(completer)
+        combobox.setEditable(True)
+        combobox.addItem("----")
+        combobox.addItems(items)
+
+        return combobox
+
+    def _get_assets(self):
+        return self._get_documents_by_name("asset")
+
+    def _get_subsets(self):
+        return self._get_documents_by_name("subset")
+
+    def _get_representations(self):
+        return self._get_documents_by_name("representation")
+
+    def _get_documents_by_name(self, document_type, parent=None):
+
+        query = {"type": document_type}
+        if parent:
+            query["parent"] = parent["_id"]
+
+        documents = io.find(query)
+        documents_by_name = set([s["name"] for s in documents])
```

Comment:
You should be able to do `documents.distinct("name")` and get all unique name entries in one go. That should be much faster, because `io.find()` should actually return a Cursor object that doesn't necessarily return all data directly as such the lookup is smaller when using distinct instead of iterating manually. :)

Reply:
Perfect, I lost this line somewhere in my notes :') I knew it was possible